### PR TITLE
Fix implementation for $L^B$ with mixed boundary conditions

### LINKED ISSUE #109: Robust unit tests and algebra fix for `L_2`
- Fix unit tests so that they also contain the case where 
  - [x] $\Delta_{0,+} \neq \Delta_{1,-}$ (Resolved by 3ce36ac)
  - [x] $\Delta_{M-1,+} \neq \Delta_{M,-}$ (Resolved by 3ce36ac)
- [x] Fix L_2 implementation based on appendix at version ece6db157741d31d00c65bc4044fc236ea3a0a30 (Resolved by b05507b)

### LINKED ISSUE #110: Formula fix and docs change on mixed boundary conditions
- [x] Fix formula for mixed boundary conditions in tex (in particular, use $1 + \underline{\xi} \Delta$ should be fixed to $(1 - \underline{\xi} \Delta)^{-1}$ and vice versa for u.b. (Resolved by #111 )
- [x] Fix implementation and unit tests accordingly (Resolved by 3ce36ac)

Note that this does not effect any reflecting barrier condition ones.

### PR DESCRIPTION
**Pull request by chiyahn**
Resolves #109 and resolves #110

### CODE REVIEW

File: src/operators.jl [hunk 1]
```diff
@@ -111,15 +111,22 @@ julia> DifferentialOperator(x̄, (Reflecting(), Reflecting()), CentralSecondDiff
...
+    Δ_1m = x̄[2] - x̄[1]
+    Δ_1p = x̄[3] - x̄[2]
+    Δ_Mm = x̄[end-1] - x̄[end-2]
+    Δ_Mp = x̄[end] - x̄[end-1]
 
     # get basis operator on interior nodes
     L = get_basis_operator(x̄, method)
-
+    
+    Ξ_1 = -2*(1/(Δ_1m*Δ_1p)-1/((Δ_1p+Δ_1m)*(Δ_1m)))
```
Comment (jlperla) [inline src/operators.jl]: Are these correct?

Comment (chiyahn) [reply src/operators.jl]: Yup, I'll create a separate issue to work on full derivation of $L_2^B$ and $L_{1+}^B$.
